src/modules/LanItems.rb

```diff
@@ -2530,6 +2523,29 @@ def clear_set_hostname
       ret
     end
 
+    # Returns unused name for device of given type
+    #
+    # When already having eht0, eth1, enp0s3 devices (eth type) and asks for new
+    # device of eth type it will e.g. return eth2 as a free name.
+    #
+    # Method always returns name in the oldfashioned schema (eth0, br1, ...)
+    #
+    # Raises an exception when type is incorrect.
+    #
+    # @param [String] device type
+    # @return [String] available device name
+    def new_type_device(type)
+      raise ArgumentError, "Valid device type expected" if type.nil? || type.empty?
+
+      known_devs = find_type_ifaces(type)
+
+      return type << "0" if known_devs.empty?
```

Comment:
1. The special case is unnecessary. Without it, the method produces the correct result, efficiently.
2. String#<< modifies the receiver. So this method mutates its argument, which happens to be `@type`!